### PR TITLE
Campaign contributors joins

### LINKED ISSUE #42: Create campaign_contributors joins table for Contributors and Campaigns
Create a campaign_contributors joins table to associate contributors with the campaigns to which they contributed.
- [x] Create contributors table with email attribute that is nullable
- [x] Create campaign_contributors joins table to create the many-to-many relationship between contributors with campaigns
- [x] Test the many-to-many relationship between campaigns and contributors

### LINKED ISSUE #46: Add value column to campaign table
Add a value column to store the value held in the campaign to the campaign table

### LINKED ISSUE #52: Campaigns by contributor endpoint
- [x] Add endpoint to grab all campaigns a contributor has contributed too
- Not sure but this may be handled by the get contributor by id endpoint

### PR DESCRIPTION
**Pull request by AntonioJacksonII**
## What's this PR do?
* Adds the contributor and campaign_contributors tables to the database
* Adds the endpoints for adding a contributor to a campaign and viewing campaigns by contributor address
## Where should the reviewer start? 
* `README.md`
* `src/models/Contributor.py`
* `src/models/CampaignContributor.py`
* `src/tests/test_campaign.py`
* `src/tests/test_contributor.py`
* `src/views/CampaignView.py`
* `src/views/ContributorView.py`
## How should this be manually tested?
* Reviewer can complete a database migration and run `pytest` to verify 19 passing tests
* Reviewer can fire up their server and hit the new endpoints to verify functionality
## Any background context you want to provide?
* FE requested joins functionality between contributors and campaigns
## What are the relevant tickets?
* Closes #46 
* Closes #42 
* Closes #52

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -396,7 +396,96 @@ To make updates to the database and run a new migration, do the following:
...
 * Example response
 ```
-{ 
+{
     "USD": 357
 }
-```
\ No newline at end of file
+```
+#### Add a Contributor to a Campaign
```
Comment (perryr16) [inline README.md:403]: Way to update the readme in real time

File: src/models/Campaign.py [hunk 2]
```diff
@@ -12,9 +14,9 @@ class Campaign(db.Model):
   description = db.Column(db.Text)
   image = db.Column(db.String)
   manager = db.Column(db.String, nullable=False)
-  contributors = db.Column(db.Integer)
   upvote = db.Column(db.Integer)
   min_contribution = db.Column(db.Float, nullable=False)
+  value = db.Column(db.Float)
```
Comment (perryr16) [inline src/models/Campaign.py:19]: - [x] value

File: src/models/CampaignContributor.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+from marshmallow import fields, Schema
+import datetime
+from . import db
+
+
+class CampaignContributor(db.Model):
```
Comment (perryr16) [inline src/models/CampaignContributor.py:6]: joins table. glad to see one of these again

File: src/models/Contributor.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+class Contributor(db.Model):
+  """
+  Contributor Model
+  """
+
+  # table name
+  __tablename__ = 'contributor'
+  id = db.Column(db.Integer, primary_key=True)
+  address = db.Column(db.String, unique=True, nullable=False)
+  email = db.Column(db.String)
```
Comment (perryr16) [inline src/models/Contributor.py:15]: looks great

File: src/tests/test_contributor.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    with self.app.app_context():
+      # create all db objects
+      db.drop_all()
+      db.create_all()
+      # delete test campaigns by name
+      existing_contributor1 = Contributor.get_contributor_by_address(self.contributor1["address"])
+      if existing_contributor1:
+        existing_contributor1.delete()
+      existing_contributor2 = Contributor.get_contributor_by_address(self.contributor2["address"])
```
Comment (perryr16) [inline src/tests/test_contributor.py:43]: I've found that `db.drop_all()` takes care of the `existing_object` check. I'm fairly sure you can delete lines 40-45

**Review by perryr16 — APPROVED** [review]
Well done, very easy to read and thanks for updating the readme

**Review by perryr16 — APPROVED** [review]

File: src/app.py [hunk 6]
```diff
@@ -26,6 +27,7 @@ def create_app(env_name):
   app.register_blueprint(campaign_blueprint, url_prefix='/api/v1/campaigns')
   app.register_blueprint(request_blueprint, url_prefix='/api/v1/requests')
   app.register_blueprint(price_converter_blueprint, url_prefix='/api/v1/price_converter')
+  app.register_blueprint(contributor_blueprint, url_prefix='/api/v1/contributor')
```
Comment (takeller) [inline src/app.py:30]: This section is getting kinda big. Possible refactor in the future? Maybe we can place the blueprint imports and blueprint app registration in its own file?

File: src/seeds/seed.py [hunk 7]
```diff
@@ -6,85 +6,82 @@
...
         # Uses lorem picsum for random pics
-        'image': 'https://picsum.photos/200/300', 
+        'image': 'https://picsum.photos/200/300',
         # Manager is a random string
         'manager': 'X5JT498FJeklnsd8382hf',
-        'contributors': 25,
-        'upvote': 50, 
-        'min_contribution': 1.5, 
-        # Address is a random string 
-        'address': 'FjSDh482hfjGE77dk',
```
Comment (takeller) [inline src/seeds/seed.py:26]: Good catch updating these seeds

**Review by takeller — APPROVED** [review]
Fat PR here. Nice work on all of this! Can't wait to integrate this in with my little mailer.

**Pull request merged**